contracts/common/Registry.sol

```diff
@@ -68,6 +73,17 @@ contract Registry is Ownable {
     proofValidatorContracts[_validator] = true;
   }
 
+  function whitelistPredicate(address predicate, uint8 _type) public onlyOwner {
+    require(predicates[predicate]._type == Type.Invalid, "Predicate already added");
```

Comment:
Not sure we need it or am I missing something?

Reply:
Changed it to
```
function addPredicate(address predicate, Type _type) public onlyOwner
  {
    require(predicates[predicate]._type == Type.Invalid, "Predicate already added");	
    predicates[predicate]._type = _type;
    emit PredicateAdded(predicate, msg.sender);
  }
```
We do need the `require` statement right?